Mentor schema update

Josh2035 opened the pull request:
## Description

#### Updates to the Mentor schema:
- email changed to EmailStr
- tech_stack changed to array (List)
- Updated MentorUpdate class to match new changes
- Updated Mentor generator class, now RandomMentor

Name changes:
- able_to_commit > commitment
- how_heard_about_us > referred_by
- anything_else > other_info (changed to Optional)

https://www.loom.com/share/51b0a264a1f748ae9c68917edc72ac78

## Useful information for testing
Sample mentor form used for testing: https://imgur.com/bYhOSnp
/create/mentor
/read/mentor
/update/mentors 

## Type of change

- [x] Bug fix

## Checklist:

- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

leila-alhemali reviewed (approved):
Next time please include more in-depth PR description. Thank you for making these changes! 

I believe these are all of the fields we need to start submitting mentor froms.

merinakarki reviewed (approved):
@BrokenShell Thank you for letting me know. 
@Josh2035 Great job but Loom video missing.

ad-victoriam-101 commented on data_generators/user_generators.py:
```diff
@@ -12,26 +13,26 @@ def __str__(self):
         self.first_name = random_first_name()
         self.last_name = choice(last_names)
-        self.email = self.first_name + "." + self.last_name + "@gmail"
+        self.email = f"{self.first_name}.{self.last_name}@gmail.com"
```
Good Use of formatted strings

ad-victoriam-101 reviewed (changes requested):
Minus loom video looks good.

BrokenShell reviewed (changes requested):
Fantastic job overall, but a loom video is required (2 minutes should do it). Also it would be good to expand the description so that anyone can quickly see what this code change does and why. I realize that I didn't give much input on this during our workshop due to time constraints. 

Looks like the video link was added right as I was writing this... AWESOME!

Josh2035 commented:
Okay. Updated the description, added the video, and tried to add some useful information for web. I believe in the screenshot the tech_stack is still formatted wrong by me ( single item when it can be a list). But hopefully this will help web with formatting. Let us know any other changes useful to the mentor form and we will add them!

BrokenShell commented:
> Okay. Updated the description, added the video, and tried to add some useful information for web. I believe in the screenshot the tech_stack is still formatted wrong by me ( single item when it can be a list). But hopefully this will help web with formatting. Thank you for all the feedback!

Perfect!

jinjahninjah reviewed (approved):
:shipit:
